docs/debug.md

```diff
@@ -0,0 +1,192 @@
+    * **operation_duration_synchronous_read (us)** - measures the duration of FUSE read() operation for the specific FUSE mountpoint, defined by the layer digest. The unit of measurement is microseconds.
+    * **synchronous_read_count** - measures how many read() operations were issued for the specific FUSE mountpoint (defined by the layer digest) to date. The  same value can be obtained from `operation_duration_synchronous_read` as the Count property.
+    * **synchronous_bytes_served** - measures the number of bytes served for synchronous reads. 
+    * **fuse_mount_failure_count** - number of times the snapshotter falls back to use a normal overlay mount instead of mounting the layer as a FUSE mount.
+    * **background_span_fetch_failure_count** - number of errors of span fetch by background fetcher
+    * **background_span_fetch_count** - number of spans fetched by background fetcher
+    * **background_fetch_work_queue_size** - number of items in the work queue of background fetcher
+    * **operation_duration_background_fetch** - time in milliseconds to complete background fetch for a layer
+    * Individual FUSE operation failure counts
+      * fuse_node_getattr_failure_count
+      * fuse_node_listxattr_failure_count
+      * fuse_node_lookup_failure_count
+      * fuse_node_open_failure_count
+      * fuse_node_readdir_failure_count
+      * fuse_file_read_failure_count
+      * fuse_file_getattr_failure_count
+      * fuse_whiteout_getattr_failure_count
+      * fuse_unknown_operation_failure_count
+
+# Common Scenarios
+
+Although creating an exhaustive list of all possible errors that may occur when using `SOCI` would not be feasible, we will attempt to outline and provide some guidance to address common scenarios that may occur both during `rpull` and throughout the lifetime of a running container. 
+
+> _Note: To allow for more verbose logging you can set the `--log-level` flag to `debug` when starting the snapshotter._
+
+## Rpull 
+
+During `rpull` the image manifest, config, and layers without zTOCs' are fetched from the remote registry. Layers that have a zTOC are mounted as a `FUSE` file system
+
+Below are a list of common error paths that may occur in this phase:
+
+### No lazy-loading
+
+If you notice that all layers are being fetched for an image or that fuse mounts are not being created for layers with a zTOC than that means that remote mounting has failed for those layers.
+
+Once you inspect the logs you should come across an error log that contains the message `failed to prepare remote snapshot` with an `error` key describing the error that was propagated up within the snapshotter.
+
+Some possible error keys include:
+
+* `skipping mounting layer <layer_digest> as FUSE mount: no zTOC for layer`
+  
+  This "error" message is not really indicative of a true error, but rather implies that the current layer does not have an associated zTOC.
```

Comment:
Not related to this PR: do we have an issue to track removing such "error" message/key in this scenario (maybe change to debug). I feel it's quite misleading to have many error logs that are not errors.

Reply:
Yeah we have an issue tracking this #361. The error key is the default key used by logrus to add some context/include the Go error, so it will still show up regardless of the logrus level unless we change the default `WithError` key to something else.